### PR TITLE
[16.0][mig] sale_quotation_number: Migration to 16.0

### PR DESCRIPTION
**Pull request by dsolanki-initos**

### CODE REVIEW

File: sale_quotation_number/__manifest__.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 {
     "name": "Sale Quotation Numeration",
     "summary": "Different sequence for sale quotations",
-    "version": "15.0.1.0.1",
+    "version": "16.0.1.0.0",
```
Comment (rousseldenis) [inline sale_quotation_number/__manifest__.py:9]: @dsolanki-initos Could you put this change in migration commit ? Thanks

Comment (dsolanki-initos) [reply sale_quotation_number/__manifest__.py:9]: Sure, @rousseldenis

File: sale_quotation_number/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model
+    def create(self, vals):
+        if self.is_using_quotation_number(vals):
+            sequence = self.env["ir.sequence"].next_by_code("sale.quotation")
+            vals["name"] = sequence or "/"
+        return super(SaleOrder, self).create(vals)
```
Comment (rousseldenis) [inline sale_quotation_number/models/sale_order.py]:
```suggestion
        return super().create(vals)
```

Comment (rousseldenis) [reply sale_quotation_number/models/sale_order.py]: Same for every instance

File: sale_quotation_number/tests/test_sale_order.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
+# Copyright 2018 Simone Rubino - Agile Business Group
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import UserError
+from odoo.tests.common import TransactionCase
+
+
+class TestSaleOrder(TransactionCase):
+    def setUp(self, *args, **kwargs):
```
Comment (rousseldenis) [inline sale_quotation_number/tests/test_sale_order.py]:
```suggestion
    @classmethod
    def setUpClass(cls, *args, **kwargs):
```

File: sale_quotation_number/tests/test_sale_order.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        super(TestSaleOrder, self).setUp()
+        self.sale_order_model = self.env["sale.order"]
```
Comment (rousseldenis) [inline sale_quotation_number/tests/test_sale_order.py]:
```suggestion
        cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))
        cls.sale_order_model = cls.env["sale.order"]
```

For test performances

**Review by rousseldenis — CHANGES_REQUESTED** [review]
Code review.

File: sale_quotation_number/tests/test_sale_order.py (hunk 4 above)
Comment (dsolanki-initos) [reply sale_quotation_number/tests/test_sale_order.py]: Hello @rousseldenis, I've done the changes as per your feedback, Kindly check it.

**Review by rousseldenis — APPROVED** [review]
Code review

File: sale_quotation_number/models/sale_order.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ def create(self, vals):
         if self.is_using_quotation_number(vals):
```
Comment (dariodelzozzo) [inline sale_quotation_number/models/sale_order.py]: This method should be converted with `@api.model_create_multi` and `vals_list` or you will see a WARNING in the log that suggests this.

Comment (dsolanki-initos) [reply sale_quotation_number/models/sale_order.py]: Hi @dariodelzozzo I've done the changes as per your feedback. Kindly review it.

File: sale_quotation_number/models/sale_order.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            if self.is_using_quotation_number(vals):
+                sequence = self.env["ir.sequence"].next_by_code("sale.quotation")
+                vals["name"] = sequence or "/"
+            return super().create(vals)
```
Comment (dariodelzozzo) [inline sale_quotation_number/models/sale_order.py]: The `return` statement must be out of the `for` loop.
I think that you can just modify the values inside the loop and after calling `super()` with `vals_list`.
Something like this:

```
    @api.model_create_multi
    def create(self, vals_list):
        for vals in vals_list:
            if self.is_using_quotation_number(vals):
                sequence = self.env["ir.sequence"].next_by_code("sale.quotation")
                vals["name"] = sequence or "/"
        return super().create(vals_list)
```

**Review by dariodelzozzo — CHANGES_REQUESTED** [review]

File: sale_quotation_number/models/sale_order.py (hunk 6 above)
Comment (dsolanki-initos) [reply sale_quotation_number/models/sale_order.py]: Hi @dariodelzozzo Done the changes

**Review by dariodelzozzo — APPROVED** [review]
Code Review

**Comment by dariodelzozzo** [comment]
@pedrobaeza This PR can be merged?
Thanks

**Pull request merged**